crates/core/src/routing_info/mod.rs

```diff
@@ -228,3 +274,59 @@ async fn sync_routing_information(
         }),
     );
 }
+
+#[cfg(any(test, feature = "test-util"))]
+pub mod test_util {
+    use std::collections::HashMap;
+    use std::ops::Deref;
+    use std::sync::{Arc, OnceLock};
+
+    use arc_swap::ArcSwap;
+    use tokio::sync::mpsc;
+
+    use crate::routing_info::PartitionRouting;
+    use restate_types::identifiers::PartitionId;
+    use restate_types::{GenerationalNodeId, NodeId, Version};
+
+    pub struct MockPartitionRouting {
+        my_node_id: GenerationalNodeId,
+        partition_routing: PartitionRouting,
+    }
+
+    impl MockPartitionRouting {
+        pub fn local_only() -> Self {
+            let (sender, _) = mpsc::channel(1);
+            let mut mappings = HashMap::default();
+
+            // Matches MockPartitionSelector's default
+            let my_node_id = GenerationalNodeId::new(0, 1);
+
+            mappings.insert(PartitionId::MIN, NodeId::Generational(my_node_id));
+            MockPartitionRouting {
+                my_node_id,
+                partition_routing: PartitionRouting {
+                    sender,
+                    partition_to_node_mappings: Arc::new(ArcSwap::new(Arc::new(
+                        super::PartitionToNodesRoutingTable {
+                            version: Version::MIN,
+                            inner: mappings,
+                        },
+                    ))),
+                    my_node_id: OnceLock::from(my_node_id),
+                },
+            }
+        }
+
+        pub fn my_node_id(&self) -> GenerationalNodeId {
+            self.my_node_id
+        }
+    }
+
+    impl Deref for MockPartitionRouting {
+        type Target = PartitionRouting;
+
+        fn deref(&self) -> &Self::Target {
+            &self.partition_routing
+        }
+    }
```

Comment:
`Deref` is a bit of two sided sword. One the one hand it lets you express `MockPartitionRouting::local_only().to_owned()` which returns a `PartitionRouting` but on the other hand it makes it really hard to follow what's happening. Somebody that isn't aware of this "implicit" conversion would be quite confused how `MockPartitionRouting::local_only().to_owned()` results in a `PartitionRouting` value. That's why `Deref` is usually used for smart pointers. See for more details https://rust-lang.github.io/api-guidelines/predictability.html#only-smart-pointers-implement-deref-and-derefmut-c-deref.